src/NLog/Layouts/Layout.cs

```diff
@@ -271,35 +268,45 @@ void ISupportsInitialize.Close()
         /// Initializes this instance.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
-		internal void Initialize(LoggingConfiguration configuration)
-		{
+        internal void Initialize(LoggingConfiguration configuration)
+        {
             if (!this.isInitialized)
             {
                 this.LoggingConfiguration = configuration;
                 this.isInitialized = true;
+                this.scannedForObjects = false;
+
+                this.InitializeLayout();
```

Comment:
this is the only thing i'm doubting about. It's now before the scan (yes I see the default scan in InitializeLayout).

I a subclass is dependent of the scan, then we could have problems if it calls base. InitializeLayout() not (or too late)?